qa/unconfigured-node-name/build.gradle

```diff
@@ -17,18 +17,15 @@
  * under the License.
  */
 
+apply plugin: 'elasticsearch.testclusters'
 apply plugin: 'elasticsearch.standalone-rest-test'
 apply plugin: 'elasticsearch.rest-test'
 
-integTestCluster {
-  setting 'node.name', null
-  // Run with no discovery configuration at all, demonstrating that a node in its
-  // "out-of-the-box" configuration can automatically bootstrap a cluster
-  autoSetInitialMasterNodes = false
-  autoSetHostsProvider = false
+testClusters.integTest {
+  nameCustomization = { null }
```

Comment:
Is there a way to do this in a more first-class way instead of using a `null` node name? The node name actually isn't relevant here. We just want to disable discovery. Perhaps we should make that a formal option on `TestClusterConfiguration`? We can then remove all the handling for null node names which is confusing IMO.

Reply:
We do also care about not configuring a node name for the test. We are also testing that we can start up without it. I taught about having a more formal way to do this, but this is the only test that would use it and it's a bit of an edge case. 
Would definitely consider extending this if we were to have other users.

Reply:
Ah ok. I assumed the empty node name was just an implementation detail.